### PR TITLE
Detailed debug

### PR DESCRIPTION
**Pull request by rb-determined-ai**
## Description

(this is a reopening of #1403 which I apparently cannot actually reopen since I force-pushed to the branch)

Presently, setting debug: true will result in log vomit. We need a way to enable specific subsystems of debug output.

This will save us lots of time in debugging customers, because we can productize some of the one-off patches we've been writing a lot of.

Additionally, this PR changes how we do logging throughout the harness. As a library, we should never be using logging.info() because that will use the root logger, and we should be using loggers specific to the subsystem which is generating the information.

### CODE REVIEW

**Comment by rb-determined-ai** [comment]
@davidhershey please review just the howto/debug.txt file.  It's still in a pretty early stage.

**Review by davidhershey — COMMENTED** [review]
I really like the flow and the overall tone.  This looks good to me -- especially once tutorials are done for the early debugging steps.

File: docs/how-to/debug.txt [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+are some simple and practical examples of what these tests might look
+like for ``PyTorchTrial`` and ``TFKerasTrial`` in the
+:meth:`determined.TrainContext.from_config` documentation, but
+ultimately only you know what a reasonable test for your Trial needs to
+look like.
+
+**How to diagnose failures:** If you hit any issues running the methods
+of your Trial code locally, there are most likely errors in your Trial
+class, or possibly in the ``hyperparameters`` section of your config
+file and your Trial class. Ideally, breaking it down method-by-method
```
Comment (rb-determined-ai) [inline docs/how-to/debug.txt:91]: "Trial class" mentioned twice.

**Comment by rb-determined-ai** [comment]
I'm closing this in light of the upcoming observability project, which is like this but much better.

The debug docs has been separated into its own change: https://github.com/determined-ai/determined/pull/1895

**Pull request closed without merging**